gwin/models/gaussian_noise.py

```diff
@@ -555,3 +555,299 @@ def _loglr(self):
+        self._margphi = kwargs.get('phase_marginalization', False)
+
+    def default_stats(self):
+        """The stats that ``get_current_stats`` returns by default."""
+        if self._margtime == True:
+            return ['logjacobian', 'logprior', 'loglr'] + \
+                   ['{}_optimal_snrsq'.format(det) for det in self._data]
+        else:
+            return ['logjacobian', 'logprior', 'loglr'] + \
+                   ['{}_optimal_snrsq'.format(det) for det in self._data] + \
+                   ['{}_matchedfilter_snrsq'.format(det) for det in self._data]
+
+    def _loglr(self):
+        r"""Computes the log likelihood ratio,
+        """
+        params = self.current_params
+        try:
+            wfs = self._waveform_generator.generate(**params)
+        except NoWaveformError:
+            return self._nowaveform_loglr()
+        opt_snr = 0.
+        mf_snr = 0j
+        logl = 0.
+        if self_margdist:
+            # assume a flat prior between 50 and 5000
+            self._mindist = 50.
+            self._maxdist = 5000.
+            dist_array = numpy.linspace(self._mindist, self._maxdist, 10**4)
+            delta_d = dist_array[1] - dist_array[0]
+        if self._margtime:
+            for det, h in wfs.items():
+                # the kmax of the waveforms may be different than internal kmax
+                kmax = min(len(h), self._kmax)
+                if self._kmin >= kmax:
+                    # if the waveform terminates before the filtering low
+                    # frequency cutoff, then the loglr is just 0 for this
+                    # detector
+                    hh_i = 0.
+                    hd_i = 0j
+                else:
+                    hh_i = h[self._kmin:kmax].inner(h[self._kmin:kmax]).real
+                    hd_i = 4. * 1/(delta_t*len(time_array)) * numpy.fft.ifft(
+                           self.data[det][self._kmin:kmax] *
+                           h[self._kmin:kmax]).real
+                opt_snr += hh_i
+                mf_snr += hd_i
+        else:
+            for det, h in wfs.items():
+                # the kmax of the waveforms may be different than internal kmax
+                kmax = min(len(h), self._kmax)
+                if self._kmin >= kmax:
+                    # if the waveform terminates before the filtering low
+                    # frequency cutoff, then the loglr is just 0 for this
+                    # detector
+                    hh_i = 0.
+                    hd_i = 0j
+                else:
+                    h[self._kmin:kmax] *= self._weight[det][self._kmin:kmax]
+                    hh_i = h[self._kmin:kmax].inner(h[self._kmin:kmax]).real
+                    hd_i = self.data[det][self._kmin:kmax].inner(
+                           h[self._kmin:kmax])
+                opt_snr += hh_i
+                mf_snr += hd_i
+        if self._margtime:
+            if self._margdist:
+                if self._margphi:
```

Comment:
I would recommend using something like this to avoid highly nested blocks:

```python
    if self._margtime and self._margdist and self._margphi:
        ...
    elif self._margtime and self._margdist:
        ...
```

and so on.